### PR TITLE
r/storagegateway_nfs_file_share - add support for `notification_policy` + validations

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_storagegateway_nfs_file_share - add support for `notification_policy`.
resource_aws_storagegateway_nfs_file_share - add plan time validation for `client_list`, `nfs_file_share_defaults. directory_mode`, `nfs_file_share_defaults. file_mode`, `nfs_file_share_defaults. group_id`, `nfs_file_share_defaults. owner_id`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
--- PASS: TestAccAWSStorageGatewayNfsFileShare_notificationPolicy (536.01s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (530.15s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (331.81s)
```

### CODE REVIEW

File: aws/internal/service/storagegateway/waiter/waiter.go [hunk 1]
```diff
@@ -28,3 +30,41 @@ func StoredIscsiVolumeAvailable(conn *storagegateway.StorageGateway, volumeARN s
 
 	return nil, err
 }
+
+// NfsFileShareAvailable waits for a NFS File Share to return Available
+func NfsFileShareAvailable(conn *storagegateway.StorageGateway, fileShareArn string, timeout time.Duration) (*storagegateway.NFSFileShareInfo, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"BOOTSTRAPPING", "CREATING", "RESTORING", "UPDATING"},
+		Target:  []string{"AVAILABLE"},
```
Comment (ewbankkit) [inline aws/internal/service/storagegateway/waiter/waiter.go:38]: Define these as constants?

File: aws/internal/service/storagegateway/waiter/waiter.go [hunk 2]
```diff
@@ -28,3 +30,41 @@ func StoredIscsiVolumeAvailable(conn *storagegateway.StorageGateway, volumeARN s
...
+		return output, err
+	}
+
+	return nil, err
+}
+
+func NfsFileShareNotFound(conn *storagegateway.StorageGateway, fileShareArn string, timeout time.Duration) (*storagegateway.NFSFileShareInfo, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending:        []string{"AVAILABLE", "DELETING", "FORCE_DELETING"},
+		Target:         []string{"NotFound"},
```
Comment (ewbankkit) [inline aws/internal/service/storagegateway/waiter/waiter.go:56]: Should work with just an empty array `[]string{}`.

File: aws/internal/service/storagegateway/waiter/waiter.go [hunk 3]
```diff
@@ -28,3 +30,41 @@ func StoredIscsiVolumeAvailable(conn *storagegateway.StorageGateway, volumeARN s
...
+	outputRaw, err := stateConf.WaitForState()
+
+	if output, ok := outputRaw.(*storagegateway.NFSFileShareInfo); ok {
+		return output, err
+	}
+
+	return nil, err
+}
+
+func NfsFileShareNotFound(conn *storagegateway.StorageGateway, fileShareArn string, timeout time.Duration) (*storagegateway.NFSFileShareInfo, error) {
```
Comment (ewbankkit) [inline aws/internal/service/storagegateway/waiter/waiter.go:53]: Rename to `NfsFileShareDeleted`?

File: website/docs/r/storagegateway_nfs_file_share.html.markdown [hunk 4]
```diff
@@ -40,6 +40,7 @@ The following arguments are supported:
 * `requester_pays` - (Optional) Boolean who pays the cost of the request and the data download from the Amazon S3 bucket. Set this value to `true` if you want the requester to pay instead of the bucket owner. Defaults to `false`.
 * `squash` - (Optional) Maps a user to anonymous user. Defaults to `RootSquash`. Valid values: `RootSquash` (only root is mapped to anonymous user), `NoSquash` (no one is mapped to anonymous user), `AllSquash` (everyone is mapped to anonymous user)
 * `file_share_name` - (Optional) The name of the file share. Must be set if an S3 prefix name is set in `location_arn`.
+* `notification_policy` - (Optional) The notification policy of the file share. For more information see [AWS Documnetation](https://docs.aws.amazon.com/storagegateway/latest/APIReference/API_CreateNFSFileShare.html#StorageGateway-CreateNFSFileShare-request-NotificationPolicy). Default value is `{}`.
```
Comment (ewbankkit) [inline website/docs/r/storagegateway_nfs_file_share.html.markdown:43]:
```suggestion
* `notification_policy` - (Optional) The notification policy of the file share. For more information see the [AWS Documentation](https://docs.aws.amazon.com/storagegateway/latest/APIReference/API_CreateNFSFileShare.html#StorageGateway-CreateNFSFileShare-request-NotificationPolicy). Default value is `{}`.
```

File: aws/resource_aws_storagegateway_nfs_file_share.go [hunk 5]
```diff
@@ -89,26 +97,28 @@ func resourceAwsStorageGatewayNfsFileShare() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"directory_mode": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "0777",
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "0777",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-7]{4}$`), ""),
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_nfs_file_share.go:103]: Maybe refactor this to a shared function in `validators.go`?

File: aws/resource_aws_storagegateway_nfs_file_share.go [hunk 6]
```diff
@@ -89,26 +97,28 @@ func resourceAwsStorageGatewayNfsFileShare() *schema.Resource {
...
+							Optional:     true,
+							Default:      "0666",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-7]{4}$`), ""),
 						},
 						"group_id": {
 							Type:         schema.TypeInt,
 							Optional:     true,
 							Default:      65534,
-							ValidateFunc: validation.IntAtLeast(0),
+							ValidateFunc: validation.IntBetween(0, 4294967294),
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_nfs_file_share.go:115]: `aws/resource_aws_storagegateway_nfs_file_share.go:115:47: constant 4294967294 overflows int`
I think this is for 32-bit targets.
See a discussion [here](https://github.com/hashicorp/terraform-provider-aws/pull/14030#pullrequestreview-441689970) on resolution.

File: aws/resource_aws_storagegateway_nfs_file_share.go (hunk 5 above)
Comment (DrFaust92) [reply aws/resource_aws_storagegateway_nfs_file_share.go:103]: good point, will do.

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayNfsFileShare_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_basic
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_basic
=== RUN   TestAccAWSStorageGatewayNfsFileShare_tags
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_tags
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ClientList
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ClientList
=== RUN   TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== RUN   TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== RUN   TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== RUN   TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== RUN   TestAccAWSStorageGatewayNfsFileShare_Squash
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_Squash
=== RUN   TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== RUN   TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== RUN   TestAccAWSStorageGatewayNfsFileShare_disappears
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_basic
=== CONT  TestAccAWSStorageGatewayNfsFileShare_disappears
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (233.49s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (238.55s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (335.94s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_Squash
--- PASS: TestAccAWSStorageGatewayNfsFileShare_notificationPolicy (341.44s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_RequesterPays
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (291.63s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ReadOnly
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (306.69s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ObjectACL
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (281.98s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (286.71s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (286.21s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (348.52s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (229.35s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (276.49s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ClientList
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (266.54s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (364.37s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_tags
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (295.96s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (312.26s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2460.046s
```

File: aws/internal/service/storagegateway/waiter/waiter.go (hunk 3 above)
Comment (DrFaust92) [reply aws/internal/service/storagegateway/waiter/waiter.go]: Done.

File: aws/internal/service/storagegateway/waiter/waiter.go (hunk 2 above)
Comment (DrFaust92) [reply aws/internal/service/storagegateway/waiter/waiter.go]: Done.

File: aws/resource_aws_storagegateway_nfs_file_share.go (hunk 5 above)
Comment (DrFaust92) [reply aws/resource_aws_storagegateway_nfs_file_share.go]: Done.

File: aws/resource_aws_storagegateway_nfs_file_share.go (hunk 6 above)
Comment (DrFaust92) [reply aws/resource_aws_storagegateway_nfs_file_share.go]: Done. Thanks for this!

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayNfsFileShare_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_basic
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_basic
=== RUN   TestAccAWSStorageGatewayNfsFileShare_tags
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_tags
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ClientList
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ClientList
=== RUN   TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== RUN   TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== RUN   TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== RUN   TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== RUN   TestAccAWSStorageGatewayNfsFileShare_Squash
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_Squash
=== RUN   TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== RUN   TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== RUN   TestAccAWSStorageGatewayNfsFileShare_disappears
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_basic
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ObjectACL
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (217.28s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_disappears
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (288.34s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (236.07s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (406.30s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_Squash
--- PASS: TestAccAWSStorageGatewayNfsFileShare_notificationPolicy (339.72s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_RequesterPays
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (284.16s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ReadOnly
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (258.34s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (285.87s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (263.26s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (295.89s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (323.53s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ClientList
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (227.39s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (373.48s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (273.43s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_tags
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (283.37s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (296.06s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2357.520s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 for the contribution!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayNfsFileShare_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_basic
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_basic
=== RUN   TestAccAWSStorageGatewayNfsFileShare_tags
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_tags
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ClientList
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ClientList
=== RUN   TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== RUN   TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== RUN   TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== RUN   TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== RUN   TestAccAWSStorageGatewayNfsFileShare_Squash
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_Squash
=== RUN   TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== RUN   TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== RUN   TestAccAWSStorageGatewayNfsFileShare_disappears
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_basic
=== CONT  TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (305.01s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_Squash
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (373.23s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_disappears
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (276.83s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (251.05s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_RequesterPays
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (288.88s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ReadOnly
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (353.15s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ObjectACL
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (265.26s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (287.25s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (303.15s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
2020/11/23 14:21:53 [DEBUG] Authorizing security group {
  Description: "Managed by Terraform",
  GroupId: "sg-0fe7b1221ab1ac6a9",
  GroupName: "terraform-2020112322215170130000001f",
  OwnerId: "187416307283",
  Tags: [{
      Key: "Name",
      Value: "tf-acc-test-2155832415105969931"
    }],
  VpcId: "vpc-046450f29017c404b"
} egress rule: []*ec2.IpPermission{{
  FromPort: 0,
  IpProtocol: "-1",
  IpRanges: [{
      CidrIp: "0.0.0.0/0"
    }],
  ToPort: 0
}}
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (341.54s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (223.99s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (270.10s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
--- PASS: TestAccAWSStorageGatewayNfsFileShare_notificationPolicy (383.88s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ClientList
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (278.67s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_tags
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (272.67s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (324.73s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2416.054s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayNfsFileShare_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_basic
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_basic
=== RUN   TestAccAWSStorageGatewayNfsFileShare_tags
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_tags
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ClientList
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ClientList
=== RUN   TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== RUN   TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== RUN   TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== RUN   TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== RUN   TestAccAWSStorageGatewayNfsFileShare_Squash
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_Squash
=== RUN   TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
=== RUN   TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== RUN   TestAccAWSStorageGatewayNfsFileShare_disappears
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_basic
=== CONT  TestAccAWSStorageGatewayNfsFileShare_disappears
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (237.99s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (273.75s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (292.65s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_notificationPolicy
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (351.67s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_Squash
--- PASS: TestAccAWSStorageGatewayNfsFileShare_notificationPolicy (353.98s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_RequesterPays
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (268.61s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ReadOnly
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (276.06s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ObjectACL
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (274.11s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (262.10s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (271.66s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (210.03s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (358.02s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (251.01s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ClientList
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (238.28s)
=== CONT  TestAccAWSStorageGatewayNfsFileShare_tags
2020/11/23 15:26:03 [DEBUG] Reading Storage Gateway NFS File Share: {
  FileShareARNList: ["arn:aws-us-gov:storagegateway:us-gov-west-1:357342307427:share/share-15FB8C63"]
}
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (339.22s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (252.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2275.138s
```

**Pull request merged**